site/en/tutorials/estimators/cnn.ipynb

```diff
@@ -0,0 +1,1014 @@
+        "Since CNNs can take a while to train, let's set up some logging so we can track\n",
+        "progress during training. We can use TensorFlow's `tf.train.SessionRunHook` to create a\n",
+        "`tf.train.LoggingTensorHook`\n",
+        "that will log the probability values from the softmax layer of our CNN. Add the\n",
+        "following to `main()`:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "S6T10kssTpdz",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "# Set up logging for predictions\n",
+        "tensors_to_log = {\"probabilities\": \"softmax_tensor\"}\n",
+        "\n",
+        "logging_hook = tf.train.LoggingTensorHook(\n",
+        "    tensors=tensors_to_log, every_n_iter=50)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "RZdtZ6JQTsmg",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "We store a dict of the tensors we want to log in `tensors_to_log`. Each key is a\n",
+        "label of our choice that will be printed in the log output, and the\n",
+        "corresponding label is the name of a `Tensor` in the TensorFlow graph. Here, our\n",
+        "`probabilities` can be found in `softmax_tensor`, the name we gave our softmax\n",
+        "operation earlier when we generated the probabilities in `cnn_model_fn`.\n",
+        "\n",
+        "Note: If you don't explicitly assign a name to an operation via the `name` argument, TensorFlow will assign a default name. A couple easy ways to discover the names applied to operations are to visualize your graph on [TensorBoard](../../guide/graph_viz.md)) or to enable the [TensorFlow Debugger (tfdbg)](../../guide/debugger.md).\n",
+        "\n",
+        "Next, we create the `LoggingTensorHook`, passing `tensors_to_log` to the\n",
+        "`tensors` argument. We set `every_n_iter=50`, which specifies that probabilities\n",
+        "should be logged after every 50 steps of training."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "brVs1dRMT0NM",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Train the Model\n",
+        "\n",
+        "Now we're ready to train our model, which we can do by creating `train_input_fn`\n",
+        "and calling `train()` on `mnist_classifier`. Add the following to `main()`:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "h-dewpleT2sk",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "# Train the model\n",
+        "train_input_fn = tf.estimator.inputs.numpy_input_fn(\n",
+        "    x={\"x\": train_data},\n",
+        "    y=train_labels,\n",
+        "    batch_size=100,\n",
+        "    num_epochs=None,\n",
+        "    shuffle=True)\n",
+        "\n",
+        "mnist_classifier.train(\n",
```

Comment:
Can we just demonstrate the logging_hook with steps=1, then turn it off for the main training? 

```
mnist_classifier.train(
    input_fn=train_input_fn,
    steps=1,
    hooks=[logging_hook])
```

```
mnist_classifier.train(
    input_fn=train_input_fn,
    steps=1000)
```

That way we avoid thousands of lines of output in the rendered page.

Reply:
Done. Moved second training run to separate code cell to improve output. Also reworked the text around this.